### PR TITLE
vSphere Docs updates

### PR DESCRIPTION
**Pull request by mylesagray**
**What this PR does / why we need it**:

Covers my contributions to the vSphere Docs and K8s effort

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

Additional to https://github.com/kubernetes/cloud-provider-vsphere/pull/181

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -98,19 +94,25 @@ and tools that are part of any Kubernetes cluster running on vSphere. If you are
 
 ### VM
 
-<to-do myles>
+A VM is an abstraction of an operating system from the phsyical machine by creating a "virtual" representation of the physical hardware the OS expects to interact with,
```
Comment (andrewsykim) [inline docs/README.md]: phsyical -> physical

File: docs/README.md [hunk 2]
```diff
@@ -98,19 +94,25 @@ and tools that are part of any Kubernetes cluster running on vSphere. If you are
...
+this includes but is not limited to CPU instruction sets, memory, BIOS, PCI buses, etc. A VM is an entirely self-contained entity and shares no components with the host OS.
+In the case of vSphere the host OS is ESXi.
```
Comment (andrewsykim) [inline docs/README.md]: Maybe a `ESXi (see below)`?

File: docs/README.md [hunk 3]
```diff
@@ -230,3 +233,434 @@ with vSphere. Below are the key integrations that are enabled by the vSphere clo
...
+
+We will be using [`flannel`](https://github.com/coreos/flannel) for pod networking in this example, so the below needs to be run on all nodes to pass [bridged IPv4 traffic to iptables chains](https://kubernetes.io/docs/setup/independent/create-cluster-kubeadm/#tabs-pod-install-4):
+
+```sh
+sudo sysctl net.bridge.bridge-nf-call-iptables=1
+```
+
+### Enabling the VMware vSphere Cloud Provider
+
+#### On the master(s)
```
Comment (andrewsykim) [inline docs/README.md:330]: Prefer:
`On the control plane(s)`

File: docs/README.md [hunk 4]
```diff
@@ -230,3 +233,434 @@ with vSphere. Below are the key integrations that are enabled by the vSphere clo
...
+
+##### Create your `vsphere.conf` file with vCenter details
+
+For reference, the vCenter configuration looks like below (you can correlate the values in the UI to the values in the `vsphere.conf` file below):
+
+![vCenter](/docs/images/vCenter.png)
+
+Edit the below command to fill in your vCenter details before running.
+
+If you don't have a folder created with your kubernetes node VMs added we can do that quickly with `govc` (note, change `vSAN-DC` to your Datacenter name in vCenter):
```
Comment (andrewsykim) [inline docs/README.md:340]: This feels like it would be hard to grasp for a new k8s admin

**Review by andrewsykim — COMMENTED** [review]
I have more comments on this PR brewing in my mind, but let's merge early and iterate on it afterwards

File: docs/README.md [hunk 5]
```diff
@@ -230,3 +233,434 @@ with vSphere. Below are the key integrations that are enabled by the vSphere clo
 ### Kubernetes Persistent Volumes
 
 ## Installing and Operating the vSphere Cloud Provider
```
Comment (andrewsykim) [inline docs/README.md:235]: Nvm, figured it's easier to iterate on this PR. 

My biggest question is that this whole section feels more like a tutorial. Thoughts on moving this to `docs/tutorials/kubernetes-on-vsphere-with-kubeadm.md` and keeping `## Installing and Operating the vSphere Cloud Provider` a bit more generic?

Comment (mylesagray) [reply docs/README.md:235]: That would work, maybe it works best as a discrete guide and this section is more of a 1) set up VMs, 2) install required components, with links to relevant sections?

Comment (andrewsykim) [reply docs/README.md:235]: Yeah, was thinking more like:
* `vsphere.conf` and `--cloud-config` points to it. 
* ensure `--cloud-provider=vsphere` is set
* etc 

And the tutorial would be a specific walk through of that

**Comment by andrewsykim** [comment]
Going to merge this early and iterate on it a bit

**Pull request merged**